lib/cc/analyzer/container.rb

```diff
@@ -0,0 +1,98 @@
+require "posix/spawn"
+
+module CC
+  module Analyzer
+    class Container
+      DEFAULT_TIMEOUT = 15 * 60 # 15m
+
+      def initialize(
+        image:,
+        name:,
+        command: nil,
+        log: NullContainerLog.new,
+        timeout: DEFAULT_TIMEOUT
+      )
+        @image = image
```

Comment:
Cool. I haven't seen much of using instance variables directly (probably because I prefer the `attr_reader` approach).

Functionally, they're the same so I don't see a big win either way.